atomic_reactor/build.py

```diff
@@ -75,18 +76,24 @@ def __init__(self, logs=None, fail_reason=None, image_id=None,
                        should be set as labels on OpenShift build
         :param skip_layer_squash: boolean, direct post-build plugins not
                                   to squash image layers for this build
+        :param oci_image_path: str, path to OCI image directory
         """
         assert fail_reason is None or bool(fail_reason), \
             "If fail_reason provided, can't be falsy"
         # must provide one, not both
-        assert bool(fail_reason) != bool(image_id), \
+        assert not (fail_reason and image_id), \
             "Either fail_reason or image_id should be provided, not both"
+        assert not (fail_reason and oci_image_path), \
+            "Either fail_reason or oci_image_path should be provided, not both"
+        assert not (image_id and oci_image_path), \
+            "Either image_id or oci_image_path should be provided, not both"
```

Comment:
Starting to wonder whether there should be an inherited `BuildFailureResult`, `BuildImageIdResult`, `BuildOciPathResult`. It's probably fine as it is, just wonder if you considered it?

Reply:
Sounds good, however I'd prefer to do it in separate PR and change it consistently across all build plugins